### PR TITLE
Add json v2 format, update misc. examples (addresses #3453 )

### LINKED ISSUE #3453: JSON data format docs pointing to old JSON parser
##### Relevant URLs
https://docs.influxdata.com/telegraf/v1.20/data_formats/input/json/

### PR DESCRIPTION
**Pull request by noramullen1**
Closes #3453 

- Add JSON v2 format doc
- Point to v2 doc in legacy version
- Update misc. examples to use v2 version

### CODE REVIEW

**Comment by noramullen1** [comment]
Reviewers -- there was a pre-existing v2 doc that I took out of draft status. It doesn't look 'new' in the diff, but it is new to the public-facing docs, so please review the whole thing. Thank you!

File: content/telegraf/v1.21/data_formats/input/json.md [hunk 1]
```diff
@@ -3,12 +3,15 @@ title: JSON input data format
...
   telegraf_1_21:
-  
+
     name: JSON input
     weight: 70
     parent: Input data formats
 ---
 
+{{% note %}}
+The following information applies to the legacy JSON input data format. For most cases, we recommend using the [JSON v2 input data format](/telegraf/v1.20/data_formats/input/json_v2/) instead.
```
Comment (sanderson) [inline content/telegraf/v1.21/data_formats/input/json.md:13]:
```suggestion
The following information applies to the legacy JSON input data format. For most cases, we recommend using the [JSON v2 input data format](/telegraf/v1.21/data_formats/input/json_v2/) instead.
```

File: content/telegraf/v1.21/data_formats/input/json.md [hunk 2]
```diff
@@ -3,12 +3,15 @@ title: JSON input data format
...
+{{% /note %}}
 
 The JSON input data format parses a [JSON][json] object or an array of objects
```
Comment (sanderson) [inline content/telegraf/v1.21/data_formats/input/json.md:16]: Should the "JSON" link go to an anchor on the page or to another URL?

**Review by sanderson — APPROVED** [review]
Two minor things and then I think it's good.

File: content/telegraf/v1.21/data_formats/input/json.md (hunk 2 above)
Comment (noramullen1) [reply content/telegraf/v1.21/data_formats/input/json.md:16]: Another URL -- this is for the legacy/v1 version, and v2 is a different file.

**Review by samhld — APPROVED** [review]
LGTM!

**Pull request merged**